internal/ecgit/ecGit.go

```diff
@@ -0,0 +1,298 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package ecgit
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/google/go-github/v45/github"
+	log "github.com/sirupsen/logrus"
+	giturls "github.com/whilp/git-urls"
+	"golang.org/x/oauth2"
+
+	"github.com/hacbs-contract/ec-cli/internal/utils"
+)
+
+var CloneRepo = cloneRepo
+var CreateAutomatedPR = createAutomatedPR
+var CreateBranch = createBranch
+var CreatePullRequest = createPullRequest
+var ApplyDiff = applyDiff
+var CommitChange = commitChange
+var PushChange = pushChange
+var NewGithubClient = newGithubClient
+
+// cloneRepo clones the specified repository.
+func cloneRepo(repoURL string, destDir string, ctx context.Context) (*git.Repository, error) {
```

Comment:
Would this be better named `cloneRepoWithAuth` or something similar since it requires creds?

Reply:
Valid point. I'll amend.